src/sushi.sol

```diff
@@ -186,19 +198,24 @@ contract SushiJoin is CropJoin {
         );
         require(
             selector == MasterChefLike.transferOwnership.selector ||
-            selector == MasterChefLike.setMigrator.selector
+            selector == MasterChefLike.setMigrator.selector ||
+            selector == MasterChefLike.set.selector
         , "SushiJoin/wrong-function");
+        if (selector == MasterChefLike.set.selector) {
```

Comment:
For `set`, shouldn't we also confirm that the `_pid` argument matches the `pid` value stored in this contract? Otherwise, an update of _any_ rewarder allows caging _any_ SushiJoin instance.

Assuming I'm looking at the correct implementation:
https://github.com/sushiswap/sushiswap/blob/master/contracts/MasterChefV2.sol#L131